toolkit/scripts/check_entangled_specs.py

```diff
@@ -147,112 +151,73 @@
     ])
 ]
 
-def check_spec_tags(base_path: str, tags: List[str], groups: List[FrozenSet]) -> Set[FrozenSet]:
-    """Returns spec sets which violate matching rules for given tags. """
-    err_groups = set()
+def check_spec_tags(base_path: str, tags: List[str], groups: List[FrozenSet], tag_expected={}) -> bool:
+    """Check if spec set violates matching rules for any of given tags. Return True/False accordingly."""
+    has_error = False
     for group in groups:
-        variants = defaultdict(set)
-
+        if verbose: 
+            print(f"Processing group: {group}")
```

Comment:
Suggestion: instead of repeating `if verbose:` what do you think about creating a wrapper `print_verbose` function?